builder/builder.go

```diff
@@ -124,6 +125,9 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 	destPathWhitelist := make(map[string]bool)
 	manifestLists := make(map[string][]manifest) // parent image -> child images
 	var mts *states.MultiTarget
+	depIndex := 0
+	imageIndex := 0
+	dirIndex := 0
```

Comment:
Indexes need to increase monotonically across all artifacts to avoid overwriting issues, so keep them out side and close on them